api/src/generate-service-worker.js

```diff
@@ -56,60 +56,64 @@ const getLoginPageContents = async () => {
   } catch (err) {
     logger.error('Error rendering login page %o', err);
     // default to returning the file
-    return [path.join(apiSrcDirectoryPath, 'templates', 'login', 'index.html')];
+    return [path.join(apiSrcDirectoryPath, 'templates', 'login', 'index.html')]; // TODO figure out if this is right
   }
 };
 
 const appendExtensionLibs = async (config) => {
   const libs = await extensionLibs.getAll();
   // cache this even if there are no libs so offline client knows there are no libs
-  config.staticFileGlobs.push('/extension-libs');
-  config.dynamicUrlToDependencies['/extension-libs'] = JSON.stringify(libs.map(lib => lib.name));
+  config.globPatterns.push('/extension-libs');
+  config.templatedURLs['/extension-libs'] = JSON.stringify(libs.map(lib => lib.name));
   libs.forEach(lib => {
     const libPath = path.join('/extension-libs', lib.name);
-    config.staticFileGlobs.push(libPath);
-    config.dynamicUrlToDependencies[libPath] = lib.data;
+    config.globPatterns.push(libPath);
+    config.templatedURLs[libPath] = lib.data;
   });
 };
 
-// Use the swPrecache library to generate a service-worker script
+// Use the workbox library to generate a service-worker script
 const writeServiceWorkerFile = async () => {
   const config = {
-    cacheId: 'cache',
-    claimsClient: true,
+    cacheId: 'cht',
+    clientsClaim: true,
     skipWaiting: true,
-    directoryIndex: false,
-    handleFetch: true,
-    staticFileGlobs: [
-      path.join(webappDirectoryPath, '{audio,img}', '*'),
-      path.join(webappDirectoryPath, 'manifest.json'),
-      path.join(webappDirectoryPath, '*.js'),
-      path.join(webappDirectoryPath, '*.css'),
-      `!${scriptOutputPath}`, // exclude service worker path
-
-      // Fonts
-      path.join(webappDirectoryPath, 'fontawesome-webfont.woff2'),
-      path.join(webappDirectoryPath, 'fonts', 'enketo-icons-v2.woff'),
-      path.join(webappDirectoryPath, 'fonts', 'NotoSans-Bold.ttf'),
-      path.join(webappDirectoryPath, 'fonts', 'NotoSans-Regular.ttf'),
-      path.join(staticDirectoryPath, 'login', '*.{css,js}'),
-      path.join(staticDirectoryPath, 'login', 'images', '*.svg'),
+    cleanupOutdatedCaches: true,
```

Comment:
This will clean up incompatible precaches that were created by older versions of Workbox. 
What happens to the old service worker cache ('sw-precache')? 
Do we need a script to delete them for this transition?

Reply:
I checked through the source and it _should_ delete sw-precache too, but I'll need to test an upgrade to be sure.

Reply:
I tested on gamma.dev and the old cache got removed as expected.